presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectSelectiveStreamReader.java

```diff
@@ -688,13 +688,13 @@ else if (isNotNull) {
             }
             dataLength = totalLength;
             data = ensureCapacity(data, totalLength);
-            offsets = ensureCapacity(offsets, totalPositions + 1);
+            offsets = ensureCapacity(offsets, totalPositions + 1, SMALL, INITIALIZE);
```

Comment:
I noticed that useBatchMode method has a lot of commented out code. Please, remove.

Reply:
> I noticed that useBatchMode method has a lot of commented out code. Please, remove.

This is fixed in the second commit "Limit batch read mode to certain conditions for SliceDirectSelectiveStreamReader"